### ISSUE TITLE
critest: add a test to validate the reason message returned for cgroup OOM-killed container

### ISSUE THREAD

**Issue by SergeyKanzhelev** [kubernetes-sigs/cri-tools#1102]
#### What would you like to be added:

Similar to e2e test in k8s (https://github.com/kubernetes/kubernetes/pull/113205), it may be a good idea to implement similar test in critest. So runtimes will keep the reason message consistent for this case.

#### Why is this needed:

To validate the behavior.

/cc @mikebrow 

/good-first-issue
/help-wanted

**Comment by saschagrunert** [comment]
Yeah this sounds like a valuable enhancement to me! :+1:

**Comment by SergeyKanzhelev** [comment]
@saschagrunert may suggest the best placing of the test. I think it should be very similar to one of the container tests https://github.com/kubernetes-sigs/cri-tools/blob/master/pkg/validate/container.go and must create a container that will exceed the limit and check the terminated reason returned by CRI be `OOMKilled`.

**Comment by saschagrunert** [comment]
Yes placing the test side by side to the existing container ones seems reasonable. 

Means we copy a scenario like this:
https://github.com/kubernetes-sigs/cri-tools/blob/f2b1f78535da6cb0c277a5006de37279109372ee/pkg/validate/container.go#L302-L333

And use an OOM container like this:
https://github.com/kubernetes/kubernetes/pull/113205/files#diff-de3ed8f1ac621fd39028dd79a971235fc85085ee175aae938f6a927b95865d18R105-R124
